src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -264,18 +266,42 @@ public void SetAll(bool value)
         ** Exceptions: ArgumentException if value == null or
         **             value.Length != this.Length.
         =========================================================================*/
-        public BitArray And(BitArray value)
+        public unsafe BitArray And(BitArray value)
         {
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
             if (Length != value.Length)
                 throw new ArgumentException(SR.Arg_ArrayLengthsDiffer);
 
-            for (int i = 0; i < m_array.Length; i++)
+            var count = m_array.Length;
+
+            switch (count)
             {
-                m_array[i] &= value.m_array[i];
+                case 3: m_array[2] &= value.m_array[2]; goto case 2;
+                case 2: m_array[1] &= value.m_array[1]; goto case 1;
+                case 1: m_array[0] &= value.m_array[0]; goto _done;
+                case 0: goto _done;
+            }
+
+            int i = 0;
+            if (Sse2.IsSupported)
+            {
+                fixed (int* leftPtr = m_array)
+                fixed (int* rightPtr = value.m_array)
+                {
+                    for (; i < count - 3; i += 4)
```

Comment:
note: I'm not sure if it has finished making the trip through CoreFX yet, but it might be good to use `i += Vector128<int>.Count` (which should be treated as a constant by the JIT).